ui/imports/shared/views/AssetsViewNew.qml

```diff
@@ -0,0 +1,331 @@
+        symbol      [string] - token's symbol e.g. "ETH" or "SNT"
+        name        [string] - token's name e.g. "Ether" or "Dai"
+        icon        [url]    - token's icon url
+        balance     [double] - tokens balance is the commonly used unit, e.g. 1.2 for 1.2 ETH, used
+                               for sorting and computing market value
+        balanceText [string] - formatted and localized balance. This is not done internally because
+                               it may depend on many external factors
+        error       [string] - error message related to balance
+
+        marketDetailsAvailable [bool]   - specifies if market datails are available for given token
+        marketDetailsLoading   [bool]   - specifies if market datails are available for given token
+        marketPrice            [double] - specifies market price in currently used currency
+        marketChangePct24hour  [double] - percentage price change in last 24 hours, e.g. 0.5 for 0.5% of price change
+
+        communityId   [string] - for community assets, unique identifier of a community, e.g. "0x6734235"
+        communityName [string] - for community assets, name of a community e.g. "Crypto Kitties"
+        communityIcon [url]    - for community assets, community's icon url
+
+        position    [int]  - if custom order available, display position defined by the user via token management
+        canBeHidden [bool] - specifies if given token can be hidden (e.g. ETH should be always visible)
+    **/
+    property var model
+
+    // enables global loading state useful when real data are not yet available
+    property bool loading
+
+    // shows/hides list sorter
+    property bool sorterVisible
+
+    // allows/disables choosing custom sort order from a sorter
+    property bool customOrderAvailable
+
+    // switches configuring right click menu
+    property bool sendEnabled: true
+    property bool swapEnabled: true
+    property bool swapVisible: true
+
+    // global market data error, presented for all tokens expecting market data
+    property string marketDataError
+
+    // formatting function for fiat currency values
+    property var formatFiat: balance => `${balance.toLocaleString(Qt.locale())} USD`
+
+    signal sendRequested(string key)
+    signal receiveRequested(string key)
+    signal swapRequested(string key)
+    signal assetClicked(string key)
+    signal communityClicked(string communityKey)
+    signal hideRequested(string key)
+    signal hideCommunityAssets(string communityKey)
+    signal manageTokensRequested
+
+    QtObject {
+        id: d
+
+        readonly property int loadingItemsCount: 25
+    }
+
+    SortFilterProxyModel {
+        id: sfpm
+
+        sourceModel: root.model
+
+        proxyRoles: [
+            // helper role for rendering section delegate
+            FastExpressionRole {
+                name: "isCommunity"
+                expression: !!communityId ? "community" : ""
```

Comment:
This role is used only internally and it's handier to have it as a string because sections are not rendered for empty strings. Thanks to that it's not necessary to wrap section component into loader. I can leave a comment for clarity here.